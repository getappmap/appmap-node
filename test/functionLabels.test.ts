import { integrationTest, readAppmap, runAppmapNode } from "./helpers";

integrationTest("mapping labeled functions", () => {
  expect(runAppmapNode("index.js").status).toBe(0);
  expect(readAppmap()).toMatchSnapshot();
});

integrationTest("mapping labeled functions - TypeScript", () => {
  expect(runAppmapNode("yarn", "exec", "ts-node", "index.ts").status).toBe(0);
  expect(readAppmap()).toMatchSnapshot();
});
