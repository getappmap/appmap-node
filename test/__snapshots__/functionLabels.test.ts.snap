// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapping labeled functions - TypeScript 1`] = `
{
  "classMap": [
    {
      "children": [
        {
          "children": [
            {
              "labels": [
                "auth",
                "mail",
              ],
              "location": "index.ts:16",
              "name": "signUp",
              "static": true,
              "type": "function",
            },
            {
              "labels": [
                "pure",
              ],
              "location": "index.ts:21",
              "name": "lambdaWithLabel",
              "static": true,
              "type": "function",
            },
            {
              "location": "index.ts:38",
              "name": "f1",
              "static": true,
              "type": "function",
            },
            {
              "location": "index.ts:42",
              "name": "f2",
              "static": true,
              "type": "function",
            },
            {
              "labels": [
                "auth",
              ],
              "location": "index.ts:5",
              "name": "signIn",
              "static": true,
              "type": "function",
            },
          ],
          "name": "index",
          "type": "class",
        },
        {
          "children": [
            {
              "labels": [
                "pure",
              ],
              "location": "index.ts:27",
              "name": "calc",
              "static": false,
              "type": "function",
            },
            {
              "location": "index.ts:32",
              "name": "setX",
              "static": false,
              "type": "function",
            },
          ],
          "name": "A",
          "type": "class",
        },
      ],
      "name": "function-labels-appmap-node-test",
      "type": "package",
    },
  ],
  "events": [
    {
      "defined_class": "index",
      "event": "call",
      "id": 1,
      "lineno": 16,
      "method_id": "signUp",
      "parameters": [],
      "path": "index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 2,
      "parent_id": 1,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 3,
      "lineno": 5,
      "method_id": "signIn",
      "parameters": [],
      "path": "index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 4,
      "parent_id": 3,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 5,
      "lineno": 38,
      "method_id": "f1",
      "parameters": [],
      "path": "index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 6,
      "parent_id": 5,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 7,
      "lineno": 42,
      "method_id": "f2",
      "parameters": [],
      "path": "index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 8,
      "parent_id": 7,
      "thread_id": 0,
    },
    {
      "defined_class": "A",
      "event": "call",
      "id": 9,
      "lineno": 32,
      "method_id": "setX",
      "parameters": [
        {
          "class": "Number",
          "name": "x",
          "value": "42",
        },
      ],
      "path": "index.ts",
      "receiver": {
        "class": "A",
        "object_id": 1,
        "value": "A { x: 1 }",
      },
      "static": false,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 10,
      "parent_id": 9,
      "thread_id": 0,
    },
    {
      "defined_class": "A",
      "event": "call",
      "id": 11,
      "lineno": 27,
      "method_id": "calc",
      "parameters": [
        {
          "class": "Number",
          "name": "a",
          "value": "2",
        },
        {
          "class": "Number",
          "name": "b",
          "value": "3",
        },
      ],
      "path": "index.ts",
      "receiver": {
        "class": "A",
        "object_id": 1,
        "value": "A { x: 42 }",
      },
      "static": false,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 12,
      "parent_id": 11,
      "return_value": {
        "class": "Number",
        "value": "87",
      },
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 13,
      "lineno": 21,
      "method_id": "lambdaWithLabel",
      "parameters": [],
      "path": "index.ts",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 14,
      "parent_id": 13,
      "return_value": {
        "class": "Number",
        "value": "42",
      },
      "thread_id": 0,
    },
  ],
  "metadata": {
    "app": "function-labels-appmap-node-test",
    "client": {
      "name": "appmap-node",
      "url": "https://github.com/getappmap/appmap-node",
      "version": "test node-appmap version",
    },
    "language": {
      "engine": "Node.js",
      "name": "javascript",
      "version": "test node version",
    },
    "name": "test process recording",
    "recorder": {
      "name": "process",
      "type": "process",
    },
  },
  "version": "1.12",
}
`;

exports[`mapping labeled functions 1`] = `
{
  "classMap": [
    {
      "children": [
        {
          "children": [
            {
              "labels": [
                "auth",
                "mail",
              ],
              "location": "index.js:17",
              "name": "signUp",
              "static": true,
              "type": "function",
            },
            {
              "labels": [
                "pure",
              ],
              "location": "index.js:22",
              "name": "lambdaWithLabel",
              "static": true,
              "type": "function",
            },
            {
              "location": "index.js:39",
              "name": "f1",
              "static": true,
              "type": "function",
            },
            {
              "location": "index.js:43",
              "name": "f2",
              "static": true,
              "type": "function",
            },
            {
              "labels": [
                "auth",
              ],
              "location": "index.js:6",
              "name": "signIn",
              "static": true,
              "type": "function",
            },
          ],
          "name": "index",
          "type": "class",
        },
        {
          "children": [
            {
              "labels": [
                "pure",
              ],
              "location": "index.js:28",
              "name": "calc",
              "static": false,
              "type": "function",
            },
            {
              "location": "index.js:33",
              "name": "setX",
              "static": false,
              "type": "function",
            },
          ],
          "name": "A",
          "type": "class",
        },
      ],
      "name": "function-labels-appmap-node-test",
      "type": "package",
    },
  ],
  "events": [
    {
      "defined_class": "index",
      "event": "call",
      "id": 1,
      "lineno": 17,
      "method_id": "signUp",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 2,
      "parent_id": 1,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 3,
      "lineno": 6,
      "method_id": "signIn",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 4,
      "parent_id": 3,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 5,
      "lineno": 39,
      "method_id": "f1",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 6,
      "parent_id": 5,
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 7,
      "lineno": 43,
      "method_id": "f2",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 8,
      "parent_id": 7,
      "thread_id": 0,
    },
    {
      "defined_class": "A",
      "event": "call",
      "id": 9,
      "lineno": 33,
      "method_id": "setX",
      "parameters": [
        {
          "class": "Number",
          "name": "x",
          "value": "42",
        },
      ],
      "path": "index.js",
      "receiver": {
        "class": "A",
        "object_id": 1,
        "value": "A { x: 1 }",
      },
      "static": false,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 10,
      "parent_id": 9,
      "thread_id": 0,
    },
    {
      "defined_class": "A",
      "event": "call",
      "id": 11,
      "lineno": 28,
      "method_id": "calc",
      "parameters": [
        {
          "class": "Number",
          "name": "a",
          "value": "2",
        },
        {
          "class": "Number",
          "name": "b",
          "value": "3",
        },
      ],
      "path": "index.js",
      "receiver": {
        "class": "A",
        "object_id": 1,
        "value": "A { x: 42 }",
      },
      "static": false,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 12,
      "parent_id": 11,
      "return_value": {
        "class": "Number",
        "value": "87",
      },
      "thread_id": 0,
    },
    {
      "defined_class": "index",
      "event": "call",
      "id": 13,
      "lineno": 22,
      "method_id": "lambdaWithLabel",
      "parameters": [],
      "path": "index.js",
      "static": true,
      "thread_id": 0,
    },
    {
      "elapsed": 31.337,
      "event": "return",
      "id": 14,
      "parent_id": 13,
      "return_value": {
        "class": "Number",
        "value": "42",
      },
      "thread_id": 0,
    },
  ],
  "metadata": {
    "app": "function-labels-appmap-node-test",
    "client": {
      "name": "appmap-node",
      "url": "https://github.com/getappmap/appmap-node",
      "version": "test node-appmap version",
    },
    "language": {
      "engine": "Node.js",
      "name": "javascript",
      "version": "test node version",
    },
    "name": "test process recording",
    "recorder": {
      "name": "process",
      "type": "process",
    },
  },
  "version": "1.12",
}
`;
